File: src/yetty/platform/android/surface.c
/* Android surface - WebGPU surface from ANativeWindow */

#include <webgpu/webgpu.h>
#include <android/native_window.h>

WGPUSurface yetty_platform_create_surface_from_window(WGPUInstance instance, ANativeWindow *window)
{
    WGPUSurfaceSourceAndroidNativeWindow android_source = {0};
    WGPUSurfaceDescriptor surface_desc = {0};

    if (!instance || !window)
        return NULL;

    android_source.chain.sType = WGPUSType_SurfaceSourceAndroidNativeWindow;
    android_source.window = window;

    surface_desc.nextInChain = &android_source.chain;

    return wgpuInstanceCreateSurface(instance, &surface_desc);
}

File: src/yetty/platform/ios/surface.m
/* iOS surface - WebGPU surface from CAMetalLayer */

#include <webgpu/webgpu.h>
#import <QuartzCore/CAMetalLayer.h>

WGPUSurface yetty_platform_create_surface_from_layer(WGPUInstance instance, CAMetalLayer *layer)
{
    if (!instance || !layer)
        return NULL;

    WGPUSurfaceDescriptorFromMetalLayer metal_desc = {0};
    metal_desc.chain.sType = WGPUSType_SurfaceDescriptorFromMetalLayer;
    metal_desc.layer = (__bridge void *)layer;

    WGPUSurfaceDescriptor surface_desc = {0};
    surface_desc.nextInChain = (WGPUChainedStruct *)&metal_desc;

    return wgpuInstanceCreateSurface(instance, &surface_desc);
}
